File: 백준/Bronze/1085. 직사각형에서 탈출/직사각형에서 탈출.js
const input = require("fs").readFileSync("/dev/stdin").toString().trim().split(" ").map(Number);

const [x, y, w, h] = input;

const dist_x = Math.min(x, w - x);
const dist_y = Math.min(y, h - y);

console.log(Math.min(dist_x, dist_y));

File: 백준/Bronze/1085. 직사각형에서 탈출/직사각형에서 탈출.py
x, y, w, h = map(int, input().split())

# x에서 왼쪽 경계까지의 거리
dist_x = min(x, w-x)
# y에서 아래쪽 경계까지의 거리
dist_y = min(y, h-y)

# x와 y의 최소 거리를 구하여 출력
print(min(dist_x, dist_y))
